### PR TITLE
Stack troubleshooting and debugging enhancements

### PR DESCRIPTION
**Pull request by robertbartel**
Adding support for remote debugging, with emphasis on using Jetbrains IDEs, to the Docker service configuration, Docker image build, and Python service package for `requestservice`.

### CODE REVIEW

File: docker/main/requestservice/Dockerfile [hunk 1]
```diff
@@ -19,10 +19,17 @@ COPY ./entrypoint.sh entrypoint.sh
...
 
+# Copy this to have access to debugging pydevd egg
+COPY --from=sources /dmod /dmod_src
+
 # Install custom and generally available packages, starting with any custom from external source image
-# Do these separately first to work around some issues
-# TODO: fix root cause for problem this works around
-RUN pip install websockets Deprecated \
+# First install the Pycharm debug helper util egg, if it is there
+# TODO: when image tagging/versioning is improved, look at keeping this in a "debug" image only
```
Comment (mattw-nws) [inline docker/main/requestservice/Dockerfile:27]: Concur, was wondering this before I noticed the comment.

**Review by mattw-nws — COMMENTED** [review]
I don't know that I can easily test this. I don't see any issues from manual review though.

File: .gitignore [hunk 2]
```diff
@@ -8,6 +8,8 @@ __pychache__/
 docker/main/base/ssh/
```
[the remark this answers is not in the document]
Comment (robertbartel) [reply .gitignore:8]: @mattw-nws, I had to go fix something else in `.gitignore`, so I addressed this also.

**Review by christophertubbs — APPROVED** [review]
I'd love a 'Debugging with pycharm for dummies' at some point, but I'm pretty happy with it.

**Pull request merged**